app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/MainNavGraph.kt

```diff
@@ -0,0 +1,263 @@
+        vm.unAuthEvent.LaunchCollection { navController.navigateAsTopLevel(Routes.AuthCode) }
+    }
+}
+
+private fun ComposeActivity.authCodeComposable(navGraphBuilder: NavGraphBuilder, navController: NavController) {
+    navGraphBuilder.composable(Routes.AuthCode) {
+        val webViewNavigator = rememberWebViewNavigator()
+
+        val vm: AuthCodeViewModel = getViewModel()
+        vm.ConnectBaseViewModel {
+            AuthCodeScreen(
+                state = it.toState { showToolbar: Boolean ->
+                    controls.title = if (showToolbar) ComposeActivity.EMPTY_TOOLBAR_TITLE else ""
+                    navIntercept = {
+                        if (vm.requestUrl.value.isNotEmpty()) {
+                            vm.requestUrl.value = ""
+                            true
+                        } else {
+                            false
+                        }
+                    }
+                },
+                navigator = webViewNavigator
+            )
+            navIntercept = {
+                webViewNavigator.canGoBack.also {
+                    if (it) webViewNavigator.navigateBack()
+                }
+            }
+        }
+
+        vm.devOptionsEvent.LaunchCollection { navController.navigate(Routes.DevOptions) }
+        vm.homeEvent.LaunchCollection { navController.navigateAsTopLevel(Routes.Home) }
+    }
+}
+
+private fun ComposeActivity.devOptionsComposable(navGraphBuilder: NavGraphBuilder, navController: NavController) {
+    navGraphBuilder.composable(Routes.DevOptions) {
+        val viewModel: DevOptionsViewModel = getViewModel()
+        viewModel.ConnectBaseViewModel {
+            DevOptionsScreen(state = it.toState())
+        }
+
+        // Only debug/internal builds allowed to show this screen. Immediately close if somehow launched on prod release build.
+        if (!activityViewModel.devOptionsEnabled) {
+            navController.navigateUp()
+        }
+
+        controls.title = stringResource(id = R.string.dev_options_title)
+    }
+}
+
+private fun ComposeActivity.homeComposable(navGraphBuilder: NavGraphBuilder, navController: NavController) {
+    navGraphBuilder.composable(Routes.Home) {
+        val viewModel: HomeViewModel = getViewModel()
+
+        viewModel.ConnectBaseViewModel {
+            HomeScreen(state = it.toState())
+        }
+
+        controls.title = stringResource(id = R.string.home_title)
+        controls.topLevel = true
+
+        viewModel.itemSelected.LaunchCollection{
+            activityViewModel.selectedRepo.value = it.repo
+            navController.navigate(Routes.RepositoryBrowser(
+                RepositoryBrowserData(repoName = it.repo.name ?: "")
+            ))
+        }
+    }
+}
+
+private fun ComposeActivity.repositoryFileComposable(navGraphBuilder: NavGraphBuilder) {
+    navGraphBuilder.composable(
+        "file?hash={hash}&path={path}&mimeType={mimeType}",
```

Comment:
@BottleRocket-Colin nit: Nice to have the named arg (`route`) used here

```suggestion
        route = "file?hash={hash}&path={path}&mimeType={mimeType}",
```